app/src/main/java/io/imont/android/sdkdemo/adapters/DeviceAttributeAdapter.java

```diff
@@ -13,50 +14,73 @@
 import io.imont.android.sdkdemo.R;
 import io.imont.cairo.EventKey;
 import io.imont.cairo.EventKeyRegistry;
+import io.imont.cairo.events.Lock;
 import io.imont.cairo.events.OnOff;
 import io.imont.mole.client.Event;
 
-import java.util.List;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+import static io.imont.cairo.events.Lock.LockValue.LOCKED;
 
 public class DeviceAttributeAdapter extends ArrayAdapter<Event> {
 
-    private final List<Event> attributes;
+    private Map<Integer, Boolean> changes = new HashMap<>();
```

Comment:
This isn't used anywhere I believe - unless you intend to copy over the rest of the "flashing when changes" logic?